packages/core/src/RenderingEngine/helpers/setDefaultVolumeVOI.ts

```diff
@@ -0,0 +1,234 @@
+  const imageIdIndex = Math.floor(imageIds.length / 2);
+  const imageId = imageVolume.imageIds[imageIdIndex];
+  const generalSeriesModule =
+    metaData.get('generalSeriesModule', imageId) || {};
+  const { modality } = generalSeriesModule;
+  const modalityLutModule = metaData.get('modalityLutModule', imageId) || {};
+
+  const numImages = imageIds.length;
+  const bytesPerImage = scalarData.byteLength / numImages;
+  const length = scalarData.length / numImages;
+
+  let type;
+
+  if (scalarData instanceof Uint8Array) {
+    type = 'Uint8Array';
+  } else if (scalarData instanceof Float32Array) {
+    type = 'Float32Array';
+  } else {
+    throw new Error('Unsupported array type');
+  }
+
+  const scalingParameters: ScalingParameters = {
+    rescaleSlope: modalityLutModule.rescaleSlope,
+    rescaleIntercept: modalityLutModule.rescaleIntercept,
+    modality,
+  };
+
+  let scalingParametersToUse;
+  if (modality === 'PT') {
+    const suvFactor = metaData.get('scalingModule', imageId);
+
+    if (suvFactor) {
+      scalingParametersToUse = {
+        ...scalingParameters,
+        suvbw: suvFactor.suvbw,
+      };
+    }
+  }
+
+  const options = {
+    targetBuffer: {
+      arrayBuffer: scalarData.buffer,
+      offset: imageIdIndex * bytesPerImage,
+      length,
+      type,
+    },
+    priority: PRIORITY,
+    requestType: REQUEST_TYPE,
+    preScale: {
+      scalingParameters: scalingParametersToUse,
+    },
+  };
+
+  // Loading the middle slice image for a volume has two scenarios, the first one is that
+  // uses the same volumeLoader which might not resolve to an image (since for performance
+  // reasons volumes' pixelData is set via offset and length on the volume arrayBuffer
+  // when each slice is loaded). The second scenario is that the image might not reach
+  // to the volumeLoader, and an already cached image (with Image object) is used
+  // instead. For the first scenario, we use the arrayBuffer of the volume to get the correct
+  // slice for the imageScalarData, and for the second scenario we use the getPixelData
+  // on the Cornerstone IImage object to get the pixel data.
+  const image = await loadAndCacheImage(imageId, options);
+
+  let imageScalarData;
+  if (!image) {
+    imageScalarData = _getImageScalarDataFromImageVolume(
+      imageVolume,
+      imageIdIndex
+    );
+  } else {
+    imageScalarData = image.getPixelData();
+  }
+
+  // Get the min and max pixel values of the middle slice
+  const { min, max } = getMinMax(imageScalarData);
+
+  return {
+    lower: min,
+    upper: max,
+  };
+}
+
+function _getImageScalarDataFromImageVolume(imageVolume, imageIdIndex) {
+  const { scalarData, imageIds, dimensions } = imageVolume;
+  const numImages = imageIds.length;
+  const bytesPerImage = scalarData.byteLength / numImages;
+
+  const numVoxels = dimensions[0] * dimensions[1] * dimensions[2];
+  const numComponents = scalarData.length / numVoxels;
```

Comment:
We aren't really dealing with the number of components for anything are we (all the logic below probably only works for 1)? I wonder how we even calculate the window level for RGB in this case?